Empty values in inline mapping result in error

# Bug Report

- **wordrow version**: `v0.5.0-beta`
- **Operating system**: Ubuntu 19.04

## Description

Providing an incorrect mapping from the CLI (i.e. with a empty string on either side of the mapping) results in a crash.

### Actual Behaviour

A program crash.

### Expected Behaviour

The mapping is not used and the program potentially halts with an error message.

## Working Example

Use the program with a CLI specified mapping such as `--map "foo,"` or `--map ",bar"`.

## Log Output

<!--
Provide the log output of the program when running the working example.
Preferably using the `--verbose` flag.
-->

```shell
$ wordrow --map "foo," inputfile.txt
panic: 1

goroutine 1 [running]:
github.com/ericcornelissen/wordrow/internal/wordmaps.(*WordMap).AddOne(0xc0000b7d78, 0x7ffc0fad0cbf, 0x3, 0x7ffc0fad0cbf, 0x0)
        /media/win/workspace/wordrow/internal/wordmaps/wordmap.go:49 +0x230
main.processInlineMapping(0x7ffc0fad0cbf, 0x4, 0xc0000b7d78, 0x1, 0x1)
        /media/win/workspace/wordrow/cmd/wordrow/wordmap.go:98 +0x189
main.processInlineMappings(0xc00005e3b0, 0x1, 0x1, 0xc0000b7d78, 0x0, 0x0)
        /media/win/workspace/wordrow/cmd/wordrow/wordmap.go:108 +0xe8
main.getWordMap(0x0, 0x0, 0x0, 0xc00005e3b0, 0x1, 0x1, 0x0, 0x0, 0x0, 0x0, ...)
        /media/win/workspace/wordrow/cmd/wordrow/wordmap.go:128 +0xba
main.run(0x0, 0x0, 0x0, 0xc00005e3c0, 0x1, 0x1, 0x0, 0x0, 0x0, 0xc00005e3b0, ...)
        /media/win/workspace/wordrow/cmd/wordrow/main.go:12 +0x81
main.main()
        /media/win/workspace/wordrow/cmd/wordrow/main.go:51 +0xf7
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -15,6 +15,7 @@ Versioning].
 
 - Fix bug preventing the usage of map files without file extension. ([#101])
 - Fix logs of incorrect level being logged. ([#107], [#108])
+- Fix crash due to empty values in the `--map` option. ([#110])
 
 ### Miscellaneous
 
@@ -130,3 +131,4 @@ Versioning].
 [#102]: https://github.com/ericcornelissen/wordrow/pull/102
 [#107]: https://github.com/ericcornelissen/wordrow/pull/107
 [#108]: https://github.com/ericcornelissen/wordrow/pull/108
+[#110]: https://github.com/ericcornelissen/wordrow/pull/110

--- a/cmd/wordrow/wordmap.go
+++ b/cmd/wordrow/wordmap.go
@@ -106,7 +106,7 @@ func processInlineMapping(mapping string, wordmap *wordmaps.WordMap) error {
 
 	from, to := values[0], values[1]
 	if stringsx.IsEmpty(from) || stringsx.IsEmpty(to) {
-		return errors.Newf("Invalid CLI defined mapping '%s'", mapping)
+		return errors.Newf("Missing value in CLI defined mapping '%s'", mapping)
 	}
 
 	wordmap.AddOne(from, to)

--- a/cmd/wordrow/wordmap.go
+++ b/cmd/wordrow/wordmap.go
@@ -105,7 +105,7 @@ func processInlineMapping(mapping string, wordmap *wordmaps.WordMap) error {
 	}
 
 	from, to := values[0], values[1]
-	if stringsx.IsEmpty(from) {
+	if stringsx.IsEmpty(from) || stringsx.IsEmpty(to) {
 		return errors.Newf("Invalid CLI defined mapping '%s'", mapping)
 	}
 

--- a/cmd/wordrow/wordmap.go
+++ b/cmd/wordrow/wordmap.go
@@ -104,7 +104,12 @@ func processInlineMapping(mapping string, wordmap *wordmaps.WordMap) error {
 		return errors.Newf("Invalid CLI defined mapping '%s'", mapping)
 	}
 
-	wordmap.AddOne(values[0], values[1])
+	from, to := values[0], values[1]
+	if stringsx.IsEmpty(from) || stringsx.IsEmpty(to) {
+		return errors.Newf("Invalid CLI defined mapping '%s'", mapping)
+	}
+
+	wordmap.AddOne(from, to)
 	return nil
 }
 

--- a/cmd/wordrow/wordmap.go
+++ b/cmd/wordrow/wordmap.go
@@ -105,7 +105,7 @@ func processInlineMapping(mapping string, wordmap *wordmaps.WordMap) error {
 	}
 
 	from, to := values[0], values[1]
-	if stringsx.IsEmpty(from) || stringsx.IsEmpty(to) {
+	if stringsx.IsEmpty(from) {
 		return errors.Newf("Invalid CLI defined mapping '%s'", mapping)
 	}
 

--- a/cmd/wordrow/wordmap_test.go
+++ b/cmd/wordrow/wordmap_test.go
@@ -189,6 +189,22 @@ func TestProcessInlineMapping(t *testing.T) {
 			t.Fatalf("Unexpected wordmap size (got %d)", wordmapSize)
 		}
 	})
+	t.Run("Empty string", func(t *testing.T) {
+		var wordmap wordmaps.WordMap
+
+		if err := processInlineMapping("foo,", &wordmap); err == nil {
+			t.Errorf("Expected no error but got one (%s)", err)
+		}
+
+		if err := processInlineMapping(",bar", &wordmap); err == nil {
+			t.Errorf("Expected no error but got one (%s)", err)
+		}
+
+		wordmapSize := wordmap.Size()
+		if wordmapSize != 0 {
+			t.Fatalf("Unexpected wordmap size (got %d)", wordmapSize)
+		}
+	})
 }
 
 func TestProcessInlineMappings(t *testing.T) {

